client/my-sites/upgrades/domain-management/controller.jsx

```diff
@@ -319,6 +319,11 @@ module.exports = {
 	},
 
 	domainManagementTransferToOtherUser( pageContext ) {
+		const selectedSite = sites.getSelectedSite();
+		if ( selectedSite && selectedSite.options.is_automated_transfer ) {
```

Comment:
Let's avoid a new SiteList usage here. We have access to the redux store with context.
```jsx
const state = pageContext.store.getState();
const siteId = getSelectedSiteId( state );
const isAutomatedTransfer = isSiteAutomatedTransfer( state, siteId );
if ( isAutomatedTransfer ) {
    const siteSlug = getSiteSlug( state, siteId );
    page.redirect( '/domains/manage/${ siteSlug }' );
}
```